### PR TITLE
Location update flag added

### PR DESCRIPTION
**Pull request by Shilpi75**
Location update flag added

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/services/FusedLocationService.java [hunk 1]
```diff
@@ -161,8 +166,11 @@ public void startLocationUpdates() {
      * Stops location updates.
      */
     public void stopLocationUpdates() {
-        Log.d(TAG, "Loction update stop.");
-        mFusedLocationClient.removeLocationUpdates(mLocationCallback);
+        if(isReceivingLocationUpdates) {
+            isReceivingLocationUpdates = false;
+            Log.d(TAG, "Loction update stop.");
+            mFusedLocationClient.removeLocationUpdates(mLocationCallback);
+        }
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/services/FusedLocationService.java]: 1. Please check if we need to do it [this](https://github.com/googlesamples/android-play-location/blob/5f83047c8a462d7c619f6275b624e219b4622322/LocationUpdates/app/src/main/java/com/google/android/gms/location/sample/locationupdates/MainActivity.java#L422) way. Maybe because it is async and can be left in an inconsistent state.
2. Place line 161 in the else case.
3. Also, fix the `task.isSuccessful` bug here only. I think we need to attach `addOnCompleteListener` and then check if success is achieved.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/services/FusedLocationService.java [hunk 2]
```diff
@@ -66,6 +68,7 @@
     private ActivityRecognitionClient mActivityRecognitionClient;
     private ActivityDetectionReceiver mActivityDetectionReceiver;
     private boolean isReceivingLocationUpdates;
+    private boolean isServiceRunning;
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/services/FusedLocationService.java]: Why is this needed? You're just using it in `onCreate` and `onDestroy`. Those are anyways the only entry and exit points. :woman_artist:

Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/services/FusedLocationService.java]: Becuase switching power saver mode calls for stop services. If it is already stopped, no need to stop things again.

Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/services/FusedLocationService.java]: Isn't onDestroy called then?
